pkg/proxy/iptables/proxier.go

```diff
@@ -157,6 +155,31 @@ type serviceInfo struct {
 	serviceLBChainName       utiliptables.Chain
 }
 
+// ClusterIP is part of ServicePort interface.
+func (info *serviceInfo) ClusterIP() string {
+	return info.clusterIP.String()
+}
+
+// Port is part of ServicePort interface.
+func (info *serviceInfo) Port() int {
+	return info.port
+}
+
+// Protocol is part of ServicePort interface.
+func (info *serviceInfo) Protocol() api.Protocol {
+	return info.protocol
+}
+
+// String is part of ServicePort interface.
+func (info *serviceInfo) String() string {
+	return fmt.Sprintf("%s:%d/%s", info.clusterIP, info.port, info.protocol)
+}
+
+// HealthCheckNodePort is part of ServicePort interface.
+func (info *serviceInfo) HealthCheckNodePort() int {
+	return info.healthCheckNodePort
+}
+
```

Comment:
add a type assert:

`var _ proxy.ServicePort = &serviceInfo{}` so it will fail if interface changes.  Same for Endpoint

Reply:
Done.